typewise-alert.c

```diff
@@ -1,71 +1,33 @@
 #include "typewise-alert.h"
-#include <stdio.h>
+
+int coolingTypeLimit[3][2] = {
+  /* PASSIVE COOLING */
+  {PASSIVE_COOLING_LOWLIMIT, PASSIVE_COOLING_HIGHLIMIT},
+  /* MED ACTIVE COOLING */
+  {MED_ACTIVE_COOLING_LOWLIMIT, MED_ACTIVE_COOLING_HIGHLIMIT},
+  /* HI ACTIVE COOLING */
+  {HI_ACTIVE_COOLING_LOWLIMIT, HI_ACTIVE_COOLING_HIGHLIMIT}
+};
 
 BreachType inferBreach(double value, double lowerLimit, double upperLimit) {
-  if(value < lowerLimit) {
-    return TOO_LOW;
-  }
-  if(value > upperLimit) {
-    return TOO_HIGH;
-  }
-  return NORMAL;
+  return checkBreach(value, lowerLimit, upperLimit);
 }
 
-BreachType classifyTemperatureBreach(
-    CoolingType coolingType, double temperatureInC) {
-  int lowerLimit = 0;
-  int upperLimit = 0;
-  switch(coolingType) {
-    case PASSIVE_COOLING:
-      lowerLimit = 0;
-      upperLimit = 35;
-      break;
-    case HI_ACTIVE_COOLING:
-      lowerLimit = 0;
-      upperLimit = 45;
-      break;
-    case MED_ACTIVE_COOLING:
-      lowerLimit = 0;
-      upperLimit = 40;
-      break;
-  }
-  return inferBreach(temperatureInC, lowerLimit, upperLimit);
+BreachType classifyTemperatureBreach(CoolingType coolingType, double temperatureInC) {
+  int lowLimit = coolingTypeLimit[coolingType][LOWLIMIT];
+  int highLimit = coolingTypeLimit[coolingType][HIGHLIMIT];
+  return inferBreach(temperatureInC, lowLimit, highLimit);
 }
 
-void checkAndAlert(
-    AlertTarget alertTarget, BatteryCharacter batteryChar, double temperatureInC) {
-
-  BreachType breachType = classifyTemperatureBreach(
-    batteryChar.coolingType, temperatureInC
-  );
-
-  switch(alertTarget) {
-    case TO_CONTROLLER:
-      sendToController(breachType);
-      break;
-    case TO_EMAIL:
-      sendToEmail(breachType);
-      break;
-  }
+void checkAndAlert( BatteryCharacter batteryChar, double temperatureInC, void (*alertType) (BreachType breachType)) {
+  BreachType breachType = classifyTemperatureBreach( batteryChar.coolingType, temperatureInC );
+  (*alertType) (breachType);
```

Comment:
Good approach to abstract the alert target dependent code 👍

Reply:
👍